File: 10303 How Many Trees.cpp
#include <iostream>
#include <vector>

int main()
{
    std::vector<long long> data(1000, 0);
    data[1] = 1;
    for (int i = 2; i < 1000; ++i)
    {
        for (int j = 1; j < i - 1; ++j)
        {
            data[i] += data[j] * data[i - j - 1];
        }
        data[i] += data[i-1]*2;
    }

    int n;
    while(std::cin >> n)
    {
        std::cout << data[n] << std::endl;
    }
    return 0;
}

File: 10303 How Many Trees.py
data = [0 for i in range(1001)]
data[1] = 1
for i in range(2,1001):
    for j in range(1, i - 1):
        data[i] += data[j] * data[i - j - 1]
    data[i] += data[i-1]*2

while True:
    try:
        print(data[int(input())])
    except :
        break

